src/test_memcachedsolution.cpp

```diff
@@ -0,0 +1,864 @@
+ * solution.
+ *
+ * Project Clearwater - IMS in the cloud.
+ * Copyright (C) 2015  Metaswitch Networks Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version, along with the "Special Exception" for use of
+ * the program along with SSL, set forth below. This program is distributed
+ * in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * The author can be reached by email at clearwater@metaswitch.com or by
+ * post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+ *
+ * Special Exception
+ * Metaswitch Networks Ltd  grants you permission to copy, modify,
+ * propagate, and distribute a work formed by combining OpenSSL with The
+ * Software, or a work derivative of such a combination, even if such
+ * copying, modification, propagation, or distribution would otherwise
+ * violate the terms of the GPL. You must comply with the GPL in all
+ * respects for all of the code used other than OpenSSL.
+ * "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+ * Project and licensed under the OpenSSL Licenses, or a work based on such
+ * software and licensed under the OpenSSL Licenses.
+ * "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+ * under which the OpenSSL Project distributes the OpenSSL toolkit software,
+ * as those licenses appear in the file LICENSE-OPENSSL.
+ */
+
+#include "gtest/gtest.h"
+
+#include "memcachedstore.h"
+#include "processinstance.h"
+
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <stdio.h>
+
+static const SAS::TrailId DUMMY_TRAIL_ID = 0x12345678;
+static const int BASE_MEMCACHED_PORT = 33333;
+static const int ASTAIRE_PORT = 11311;
+
+// Fixture for all memcached solution tests.
+//
+// This fixture:
+// - Creates a unique key and an instance of TopologyNeutralMemcachedStore for
+//   each test.
+// - Tidies up instances of memcached and Astaire and any config files between
+//   sets of tests.
+// - Provides helper methods for memcached operations and managing instances of
+//   memcached and Astaire
+class BaseMemcachedSolutionTest : public ::testing::Test
+{
+public:
+  /// Kill and clear all the memcached and Astaire instances, and remove the
+  /// cluster_settings file.
+  static void TearDownTestCase()
+  {
+    for (std::vector<std::shared_ptr<MemcachedInstance>>::iterator inst = _memcached_instances.begin();
+         inst != _memcached_instances.end();
+         ++inst)
+    {
+      (*inst)->kill_instance();
```

Comment:
Should the `ProcessInstance` destructor call `kill_instance`. Then this code just becomes `_memcached_instances.clear()`